sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -76,7 +77,8 @@ class HadoopTableReader(
   override def makeRDDForTable(hiveTable: HiveTable): RDD[Row] =
     makeRDDForTable(
       hiveTable,
-      relation.tableDesc.getDeserializerClass.asInstanceOf[Class[Deserializer]],
+      Class.forName(relation.tableDesc.getSerdeClassName, true, sc.sparkContext.getClassLoader)
```

Comment:
After looking the code http://www.grepcode.com/file/repository.cloudera.com/content/repositories/releases/org.apache.hive/hive-exec/0.13.1-cdh5.3.0/org/apache/hadoop/hive/ql/session/SessionState.java#61

Probably we don't need to create the new class loader by ourselves, we can reuse the SessionState code instead.